TFNetwork.py

```diff
@@ -1371,6 +1371,10 @@ def save_params_to_file(self, filename, session):
     """
     import os
     filename = os.path.abspath(filename)  # TF needs absolute path
+    dirname = os.path.dirname(filename)
+    from Util import maybe_create_folder
+    maybe_create_folder(dirname)
+
```

Comment:
Don't randomly add empty lines. Also, avoid creating too much local variables (here: `dirname`), esp if it is only used once. You can directly use `os.path.dirname(filename)`.